actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -61,129 +66,192 @@ public void onReceive(Request request) throws Exception {
       sender().tell(response, self());
 
     } else if (request.getOperation().equalsIgnoreCase(ActorOperations.GET_CONTENT.getValue())) {
-      String userId = (String) request.getRequest().get(JsonKey.USER_ID);
+
       Response res = new Response();
-      if (request.getRequest().get(JsonKey.COURSE) != null) {
-        res = getContentByCourse(userId, request.getRequest());
-      } else if (request.getRequest().get(JsonKey.CONTENT_IDS) != null) {
-        res = getContentByContents(userId, request.getRequest());
-      } else if (request.getRequest().get(JsonKey.COURSE_IDS) != null) {
-        res = getContentByCourses(userId, request.getRequest());
+      Map<String, Object> requestMap = request.getRequest();
+      String userId = (String) request.getRequest().get(JsonKey.USER_ID);
+
+      if (StringUtils.isBlank(userId)) {
+        throw new ProjectCommonException(
+            ResponseCode.mandatoryParamsMissing.getErrorCode(),
+            ResponseCode.mandatoryParamsMissing.getErrorMessage(),
+            ResponseCode.CLIENT_ERROR.getResponseCode(),
+            JsonKey.USER_ID);
       }
+
+      res = getCourseContentState(userId, requestMap);
       removeUnwantedProperties(res);
       sender().tell(res, self());
     } else {
       onReceiveUnsupportedOperation(request.getOperation());
     }
   }
 
-  /**
-   * Method to get content on basis of course ids.
-   *
-   * @param userId String
-   * @param request Map<String, Object>
-   * @return Response
-   */
-  @SuppressWarnings("unchecked")
-  private Response getContentByCourses(String userId, Map<String, Object> request) {
+  private Response getCourseContentState(String userId, Map<String, Object> requestMap) {
 
     Response response = new Response();
-    Util.DbInfo dbInfo = Util.dbInfoMap.get(JsonKey.LEARNER_CONTENT_DB);
-    CopyOnWriteArrayList<String> courseIds =
-        new CopyOnWriteArrayList<String>((List<String>) request.get(JsonKey.COURSE_IDS));
-    List<Map<String, Object>> modifiedContentList = new ArrayList<Map<String, Object>>();
+    List<Map<String, Object>> contentList = new ArrayList<>();
 
-    LinkedHashMap<String, Object> queryMap = new LinkedHashMap<String, Object>();
-    queryMap.put(JsonKey.USER_ID, userId);
+    String batchId = (String) requestMap.get(JsonKey.BATCH_ID);
+    List<String> courseIds = new ArrayList<>();
```

Comment:
No need to define empty array lists. Add null checks instead of size checks where required.